nixos/modules/services/web-apps/convos.nix

```diff
@@ -0,0 +1,80 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.convos;
+in
+{
+  options.services.convos = {
+    enable = mkEnableOption "Convos";
+    package = mkOption {
+      default = pkgs.convos;
+      defaultText = "pkgs.convos";
+      type = types.package;
+      description = "Convos package to use";
+    };
+    listenPort = mkOption {
+      type = types.int;
+      default = 3000;
+      example = 8080;
+      description = "Port the web interface should listen on";
+    };
+    listenAddress = mkOption {
+      type = types.str;
+      default = "*";
+      example = "127.0.0.1";
+      description = "Address or host the web interface should listen on";
+    };
+    reverseProxy = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Enables reverse proxy support. This will allow Convos to automatically
+        pick up the X-Forwarded-For and X-Request-Base HTTP headers set in your
+        reverse proxy web server. Note that setting this environment variable
+        without a reverse proxy in front will be a security issue.
+      '';
+    };
+  };
+
+
+  config = mkIf cfg.enable {
+    systemd.services.convos = {
+      description = "Convos Service";
+      wantedBy = [ "multi-user.target" ];
+      after = [ "networking.target" ];
+      serviceConfig = {
+        User = "convos";
```

Comment:
Does this actually need a permanent user account, or would `DynamicUser` suffice?

Reply:
Good point. I think it should work, and makes some of the other hardening options redundant. I'll need to test it a bit and update the commit.

Reply:
Works well - I've tweaked the Protect* and Restrict* options a bit.